tests/e2e/e2e_precompile_genesis_test.go

```diff
@@ -0,0 +1,171 @@
+)
+
+// TestPrecompileGenesis tests that the the following is true for enabled precompiles:
+//
+//   - An enabled precompile has an EthAccount with matching code hash,
+//     sequence of 1, and no public key
+//   - An enabled precompile has code equal to 0x01
+//   - An enabled precompile has a nonce of 1
+//
+// This is important to ensure the genesis setup for precompiles is correct.
+func (suite *IntegrationTestSuite) TestPrecompileGenesis() {
+	type fixture struct {
+		address            string
+		expectIsEnabled    bool
+		expectIsEthAccount bool
+		expectCode         []byte
+		expectNonce        uint64
+	}
+
+	// enabled represnets the expected state for a registered precompile
+	// that is enabled at genesis
+	enabled := func(address string) func() fixture {
+		return func() fixture {
+			return fixture{
+				address:            address,
+				expectIsEnabled:    true,
+				expectIsEthAccount: true,
+				expectCode:         []byte{0x01},
+				expectNonce:        uint64(1),
+			}
+		}
+	}
+
+	// disabled represents the expected state for a registered precompile
+	// that is not enabled at genesis
+	disabled := func(address string) func() fixture {
+		return func() fixture {
+			return fixture{
+				address:            address,
+				expectIsEnabled:    false,
+				expectIsEthAccount: false,
+				expectCode:         []byte{},
+				expectNonce:        uint64(0),
+			}
+		}
+	}
+
+	testCases := []struct {
+		name       string
+		genFixture func() fixture
+	}{
+		{
+			name:       "noop contract address is enabled and initialized",
+			genFixture: enabled(registry.NoopContractAddress),
+		},
+		{
+			name:       "noop contract address second address is disabled and not initialized",
+			genFixture: disabled(registry.NoopContractAddress2),
+		},
+	}
+
+	for _, tc := range testCases {
+		suite.Run(tc.name, func() {
+			tf := tc.genFixture()
+
+			//
+			// Addresses
+			//
+			evmAddress := common.HexToAddress(tf.address)
+			sdkAddress := sdk.AccAddress(evmAddress.Bytes())
+
+			//
+			// Heights
+			//
+			// We ensure all queries happen at block 1 after genesis
+			// and help ensure determisitc behavior
+			genesisHeight := big.NewInt(1)
+			grpcGenesisContext := suite.Kava.Grpc.CtxAtHeight(genesisHeight.Int64())
+
+			//
+			// Queries
+			//
+			evmParamsResp, err := suite.Kava.Grpc.Query.Evm.Params(grpcGenesisContext, &evmtypes.QueryParamsRequest{})
+			suite.Require().NoError(err)
+
+			// accountErr is checked during assertions if we are expecting an account due to
+			// the service returning a not found error for precompiles that are not enabled
+			accountResponse, accountErr := suite.Kava.Grpc.Query.Auth.Account(
+				grpcGenesisContext, &authtypes.QueryAccountRequest{Address: sdkAddress.String()})
+			var account authtypes.AccountI
+			if accountErr == nil {
+				err = suite.Kava.EncodingConfig.Marshaler.UnpackAny(accountResponse.Account, &account)
+				suite.Require().NoError(err)
+			}
```

Comment:
```
accountResponse, accountErr := suite.Kava.Grpc.Query.Auth.Account(
grpcGenesisContext, &authtypes.QueryAccountRequest{Address: sdkAddress.String()})
if (accountErr != nil) != tf.expectIsEthAccount {
    suite.T().Errorf("expected account query to not error for address %s", tf.address)
    return
}

var account authtypes.AccountI
suite.Require().NoError(suite.Kava.EncodingConfig.Marshaler.UnpackAny(accountResponse.Account, &account))
```
little easier to read then having to jump down to see where err is being handled.

Reply:
Resolved the other similar comments - I explored the different variations here and opted to improve the comments.  This current organization is the most clear imo and is organized based on a "arrange, act, assert" approach.  With this e2e test, we focus on building a view of the blockchain state at block 1, then making assertions on that state.  There are probably better ways to make it more clear, but from that perspective moving the business logic type assertions up into the queries is a bit distracting.

The goal of the error assertions in the queries is to halt the test if we find data unavailable, which signals the state, setup, etc is invalid and running assertions does not make sense -- hence the Require() on these.

Reply:
Also, with editors it is quite easy to jump to where variables are used, so imo the weight on query & assertion organization is more important.